pages/coaching-dashboard.tsx

```diff
@@ -1,27 +1,36 @@
 import { useQuery } from "@apollo/client";
-import React from "react";
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import ProfileDetailCard from "../components/coding/studentPortal/ProfileDetailCard";
 import {
   FetchUserProfileCardResponse,
   FETCH_USER_PROFILE_CARD,
+  Users,
 } from "../graphql/fetchUserProfileCard";
 import { format } from "date-fns";
 import Link from "next/link";
+import {
+  skillifyUsersSelector,
+  setUserList,
+} from "../redux/skillifyUsersSlice";
 const coachingDashboard = () => {
+  const dispatch = useDispatch();
+  const userListState = useSelector(skillifyUsersSelector);
   const { loading, data } = useQuery<FetchUserProfileCardResponse>(
-    FETCH_USER_PROFILE_CARD
+    FETCH_USER_PROFILE_CARD,
+    {
+      onCompleted: () => dispatch(setUserList(data.users)),
+    }
   );
-
   if (loading) {
     return <div className="flex place-content-center">"Loading..."</div>;
   }
-
   return (
     <div className="flex flex-col p-4 m-4 ">
       <p className="mb-8 text-3xl">Coaching Dashboard</p>
       <h2 className="mb-4">Enrolled Students</h2>
       <div className="grid grid-cols-1 gap-4 sm:grid-cols-3">
-        {data.users.map((it) => (
+        {userListState.userList?.map((it) => (
```

Comment:
instead of calling the state object, then the state variable, you can simplify this by using object destructuring above when importing from the selector

Reply:
oh that makes sense, will do!